navbar: Channel description popover for touchscreens.

HyperDanisH opened the pull request:
<!-- Describe your pull request here.-->

Fixes: #33147 
Related to: #37188 

This PR is an attempt to solve the problem on mobile devices where the channel description is hidden due to broken design. It adds a popover support for navbar to display channel description.
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<table>
  <tr>
    <td>Closed</td>
    <td>Opened</td>
  </tr>
  <tr>
    <td>
<img width="468" height="692" alt="Screenshot 2026-01-19 at 11 52 45 PM" src="https://github.com/user-attachments/assets/f87a7d59-0540-4bac-b54e-7db42fb79612" />


</td>
    <td>
<img width="460" height="653" alt="Screenshot 2026-01-05 at 8 21 42 PM" src="https://github.com/user-attachments/assets/5f62f25c-f495-4caf-82fe-0c07f84cab63" />
</td>
  </tr>
</table>


[chrome-capture-2026-01-19.webm](https://github.com/user-attachments/assets/b6407250-7f7e-47f8-a671-c9d664602b9b)



**Remaining decisions and concerns.**
1 ) The design for this popover is inspired by the previous work done on this codebase but it might not resemble it completely; Work needs to be done here.
2 ) I studied the previous codebase and found a popover implementation using `tippy.js` library but was unsuitable for this usecase, so I used `jquery API` to implement this popover just as search area is implemented.

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! Please add still screenshots demonstrating your changes.

HyperDanisH commented:
<table>
  <tr>
    <td>
      <strong>Updated Screenshot</strong>
    </td>
  </tr>
  <tr>
    <td>
      <img
        width="474"
        height="697"
        alt="Screenshot 2026-01-05 at 1:26 AM"
        src="https://github.com/user-attachments/assets/9a6bd6f0-9f56-4a44-961c-32df76f33bc3"
      />
    </td>
  </tr>
</table>

HyperDanisH commented:
@alya I have mentioned my decisions in the PR description. Please check them out.

Thanks.

apoorvapendse commented on web/src/message_view_header.ts:
```diff
@@ -234,6 +240,77 @@ export function initialize(): void {
+                    <span class="sub-stream-description rendered_markdown content-container">
+                        ${context.description}
+                    </span>
+                </div>
```
We never use this pattern in the codebase, you can do something like:
https://github.com/zulip/zulip/blob/0bda22711a454a4cc6696ade12da0a2ca7950f85/web/src/compose_closed_ui.ts#L243-L250

HyperDanisH commented:
@apoorvapendse can you please respond to my comments?

apoorvapendse commented:
No worries, GitHub has a weird way of managing reviews and replies to comments.
Taking a look at your comments.

apoorvapendse commented on web/styles/zulip.css:
```diff
@@ -396,6 +396,80 @@ p.n-margin {
+        /* (header-height + 6px) in order to match the navbar
+        text in message_view_header */
+        padding: 12px calc(var(--header-height) + 6px);
+        hyphens: auto;
```
Cool, I think you can initiate it on the CZO thread we've been using so far.

HyperDanisH committed:
navbar: Channel description popover for touchscreens.

This commit adds support for expanding the channel description on
devices with small viewports.

It introduces:
1) notepad-text icon
2) popover for showing description.

Each time the the icon is clicked, the popover opens and
displays the full description.

Fixes zulip#33147

apoorvapendse reviewed (commented):
Just squash both the commits.
I misread the early return check as something that was there since the introduction of the container query to hide the narrow description, but looks like it was introduced as a part of this PR (we don't need it as we've established in the [discussion]( https://github.com/zulip/zulip/pull/37277#discussion_r2727495469)).


The first commit is misleading because the `hide_stream_settings_button_width_threshold` checks and logic is  still valid in that we only make it stale after this work, so we should do the related removal in the same commit.

apoorvapendse reviewed (commented):
There is a bug in case of very long words, they overflow:
<img width="689" height="165" alt="Image" src="https://github.com/user-attachments/assets/17df0072-76dd-47de-ad9f-250bf620d426" />

This is what it looks like in the extended description on larger widths (as expected):
<img width="827" height="218" alt="Image" src="https://github.com/user-attachments/assets/02fd2f9a-83fd-43ba-ac30-e43859257415" />


A similar bug exists in the channel overlay, we might want to fix that as a follow-up:
<img width="848" height="673" alt="Image" src="https://github.com/user-attachments/assets/7e246ef0-ff93-4639-940b-868d0981ca3e" />

I think this can be addressed by using a `overflow-wrap: break-word;`

apoorvapendse reviewed (commented):
You should move your css changes to `message_view_header.css`.

apoorvapendse reviewed (commented):
There is a bug with the newly introduced change of not hiding the navbar with the overlay; try keeping the overlay open and switch to a different channel, the older overlay persists.

We should probably hide it *iff* we decide to go in that direction at all.

apoorvapendse commented:
Yes, that is what I intended to suggest with

>We should probably hide it iff we decide to go in that direction at all.

apoorvapendse commented on web/styles/message_view_header.css:
```diff
@@ -139,6 +139,74 @@
+    z-index: 10000000000;
+
+    &.expanded {
+        display: block !important;
```
Let's avoid using `!important` unless we absolutely need to.

apoorvapendse commented on web/styles/message_view_header.css:
```diff
@@ -139,6 +139,74 @@
+
+    .content-container {
+        display: block;
+        padding: 12px 40px 12px 12px;
```
We should probably be using `em` units; the right padding looks excessive at smaller font sizes.

apoorvapendse reviewed (commented):
This doesn't handle vertical overflow by making the overlay scrollable.
Try bumping up the font size and enter a long channel description on a small viewport.

Otherwise, this looks in good shape.
I'd like @karlstolley to do an audit of the CSS once we are sure about the UI decisions with this PR.

apoorvapendse commented on web/src/message_view_header.ts:
```diff
@@ -224,16 +218,34 @@ export function initialize(): void {
+        }
+
+        const $el = $(render_channel_description_popover);
+        $("#header-container").append($el);
```
Please rename the variables to be consistent with the render_ function(s) we are calling.
Also, let's give $el a more descriptive name, maybe `$extended_narrow_description_overlay` in the interest of keeping it straightforward.

As a general guideline, we should try to update everything that is related to a change like renaming the associated template here; the current version of this block is quite hard to follow when reading out of context.
You can move the 
```
const render_channel_description_popover =
            render_extended_narrow_description_overlay(context);
```
line just before the place where you are appending it to the $header_container.

apoorvapendse commented on web/styles/message_view_header.css:
```diff
@@ -1,23 +1,9 @@
     /* TODO: Remove the `.navbar_title` span from the `message_view_header.hbs`
-       template. That span predates the use of flexbox, and makes it so that
-       elements like the `.narrow_description` have different line-height and
-       sizing contexts depending on whether it's a view or a narrow being shown. */
```
Let's avoid introducing spacing changes that pollute the diff.

apoorvapendse reviewed (commented):
This looks nice, left a couple minor comments.

@alya would you like to give this a try?

alya commented:
@HyperDanisH You can ping me for a review once @apoorvapendse 's feedback above has been addressed.

apoorvapendse commented on web/templates/extended_narrow_description_overlay.hbs:
```diff
@@ -0,0 +1,27 @@
+                {{> navbar_icon_and_title . }}
+            </span>
+        {{/if}}
+        <button type="button" id="extended-narrow-description-overlay-exit" aria-label="{{t 'Exit description' }}"><i class="zulip-icon zulip-icon-close" aria-hidden="true"></i></button>
```
> For reference see this x on notification section.

Interesting.

I do see one for the compose box close button, tooltips are anyways going to be less useful since we are on mobile widths.

<img width="350" height="267" alt="image" src="https://github.com/user-attachments/assets/018cdda0-85c7-4b95-95bc-df1d18ebf0ae" />

alya commented:
@HyperDanisH Are you continuing the work here, or should @apoorvapendse take it from here?

HyperDanisH commented:
Let apprrva take this. I'll be contributing but some other issues

On Thu, 21 May, 2026, 00:20 Alya Abbott, ***@***.***> wrote:

> *alya* left a comment (zulip/zulip#37277)
> <https://github.com/zulip/zulip/pull/37277#issuecomment-4501601576>
>
> @HyperDanisH <https://github.com/HyperDanisH> Are you continuing the work
> here, or should @apoorvapendse <https://github.com/apoorvapendse> take it
> from here?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/37277#issuecomment-4501601576>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWH45N77Z55Z4LQA7SMASRD43X46TAVCNFSM6AAAAACQNC7KNCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DKMBRGYYDCNJXGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
